File: plugin/zeppelin/src/main/scala/org/intellij/plugin/zeppelin/service/execution/ExecutionHandler.scala
package org.intellij.plugin.zeppelin.service.execution

import java.util.concurrent.atomic.AtomicBoolean

trait ExecutionHandler {
  def onError(msg: ExecutionResults)

  def onOutput(data: OutputResponse, isAppend: Boolean)

  def onProgress(percentage: Double): Unit

  def onSuccess(msg: ExecutionResults)

  def onUpdateExecutionStatus(status: String): Unit

  def isCompleted: AtomicBoolean
}

trait ExecutionHandlerFactory {
  def create():ExecutionHandler
}

File: zeppelin/src/org/intellij/plugin/zeppelin/service/execution/ExecutionHandler.kt
package org.intellij.plugin.zeppelin.service.execution

import java.util.concurrent.atomic.AtomicBoolean

interface ExecutionHandler {
    fun onError(msg: ExecutionResults)
    fun onOutput(data: OutputResponse, isAppend: Boolean)
    fun onProgress(percentage: Double)
    fun onSuccess(msg: ExecutionResults)
    fun onUpdateExecutionStatus(status: String)
    fun isCompleted(): AtomicBoolean
}

interface ExecutionHandlerFactory {
    fun create(): ExecutionHandler
}